### PR TITLE
[CUDA][CUBLAS] Fix BF16 reduced precision reduction note in Numerical accuracy docs

### LINKED ISSUE #100966: Inconsistent doc and code: matmul bf16 reduction setting
### 🐛 Describe the bug

In the [docs on reduced precision bf16 gemm](https://pytorch.org/docs/main/notes/cuda.html#reduced-precision-reduction-in-bf16-gemms), it is stated that:
> Note that this switch is set to *False* by default for BF16 as we have observed numerical instability in PyTorch CI tests (e.g., test/test_matmul_cuda.py).

However the actual behaviour is *True*:
```python
import torch
print(torch.backends.cuda.matmul.allow_bf16_reduced_precision_reduction)  # True
```

### Versions

PyTorch version: 2.0.0+cu117
Is debug build: False
CUDA used to build PyTorch: 11.7
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.6 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.26.3
Libc version: glibc-2.31

Python version: 3.8.16 (default, Mar  2 2023, 03:21:46)  [GCC 11.2.0] (64-bit runtime)
Python platform: Linux-5.15.0-52-generic-x86_64-with-glibc2.17
Is CUDA available: True
CUDA runtime version: 11.7.99
CUDA_MODULE_LOADING set to: LAZY

cc @svekars @carljparker

### PR DESCRIPTION
**Pull request by lkct**
Fixes #100966

Ref #101044

Align implementation and documentation. (This is what's previously missed from the above issue and PR)

cc @ngimel @ezyang @eqy

### CODE REVIEW

**Review by eqy — APPROVED** [review]
LGTM, thanks for the fix!

**Review by ezyang — APPROVED** [review]
